services/motion/builtin/builtin.go

```diff
@@ -231,24 +236,111 @@ func (ms *builtIn) MoveOnGlobe(
 	kinematicsOptions.GoalRadiusMM = math.Min(motionCfg.PlanDeviationM*1000, 3000)
 	kinematicsOptions.HeadingThresholdDegrees = 8
 
-	plan, kb, err := ms.planMoveOnGlobe(
-		ctx,
-		componentName,
-		destination,
-		movementSensorName,
-		obstacles,
-		kinematicsOptions,
-		extra,
-	)
-	if err != nil {
-		return false, fmt.Errorf("error making plan for MoveOnMap: %w", err)
-	}
+	movementSensor, ok := ms.movementSensors[movementSensorName]
+	if !ok {
+		return false, resource.DependencyNotFoundError(movementSensorName)
+	}
+
+	// TODO: if we disentangle creation of the kinematic base from the plan call we make we don't need this and can do this once
+	var kinematicBase kinematicbase.KinematicBase = nil
+
+	// Shared variables for planning and execution threads, are all guarded with a mutex
+	var planMu sync.Mutex
+	var moveCtx context.Context
```

Comment:
When do we set this?

Reply:
I was originally using `moveCtx` as a separate context than the ones we use for polling.  But then I had the realization that I was overcomplicating things, and every time we call a cancel on any one of the polling threads or move threads we actually just want to cancel all of them.  

I can see this changing in the future where we move to replanning without stopping.  But for the time being I would rather keep things simple and let a future implementer make that change.

Reply:
Ah ok. So will this be removed?

Reply:
Yes, I just pushed up an update to show what I was thinking.  I think it should be a little easier to reason about with fewer contexts